62.CAD/vertex_shader.hlsl

```diff
@@ -417,19 +483,67 @@ PSInput main(uint vertexID : SV_VertexID)
         }
 
         outV.position.xy = (outV.position.xy / globals.resolution) * 2.0 - 1.0;
+    }
+    else if (objType == ObjectType::CURVE_BOX)
+    {
+        outV.setColor(lineStyle.color);
+        outV.setLineThickness(screenSpaceLineWidth / 2.0f);
+
+        CurveBox curveBox;
+        curveBox.aabbMin = vk::RawBufferLoad<double2>(drawObj.geometryAddress, 8u);
+        curveBox.aabbMax = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2), 8u);
+        for (uint32_t i = 0; i < 3; i ++)
+        {
+            curveBox.curveMin[i] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * (2 + i), 8u);
+            curveBox.curveMax[i] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * (2 + 3 + i), 8u);
+        }
 
+        const double2 ndcAabbExtents = double2(
+            length(abs(transformVectorNdc(clipProjectionData.projectionToNDC, double2(curveBox.aabbMax.x, curveBox.aabbMin.y) - curveBox.aabbMin))),
+            length(abs(transformVectorNdc(clipProjectionData.projectionToNDC, double2(curveBox.aabbMin.x, curveBox.aabbMax.y) - curveBox.aabbMin)))
+        );
+        // Anti-alising factor + 1px due to aliasing with the bbox (conservatively rasterizing the bbox, otherwise
+        // sometimes it falls outside the pixel center and creates a hole in major axis)
+        // Doubling the factor for both the left/top and right/bottom sides
+        const double2 dilatedAabbExtents = ndcAabbExtents + 2.0 * ((globals.antiAliasingFactor + 1.0) / double2(globals.resolution));
+        double2 maxCorner = double2(bool2(vertexIdx & 0x1u, vertexIdx >> 1));
```

Comment:
a little comment here would help to describe what the maxCorner value will look like